lib/rules/jsx-classname-namespace.js

```diff
@@ -117,14 +113,15 @@ var rule = module.exports = function( context ) {
 		return nodeA.name === nodeB.name;
 	}
 
-	function isRootRenderedElement( node, filename ) {
+	function isFolderRootFile( filename ) {
+		const stripedFileName = filename.substr( filename.lastIndexOf( '/' ) + 1 );
```

Comment:
fixed (and it's already used in the rule so you can add "consistency" to the list of reasons it's a better solution :) )